src/AreaChart/AreaSeries/AreaSeries.tsx

```diff
@@ -116,140 +118,143 @@ interface AreaSeriesState {
+  const getColor = (point, index: number) => {
+    const { activeValues } = areaState;
     const key = Array.isArray(point) ? point[0].key : point?.key;
 
-    return getColor({
+    return getColorCommon({
       data,
       colorScheme,
       active: activeValues,
       point,
       index,
       key
     });
-  }
+  };
 
-  onValueEnter(event: TooltipAreaEvent) {
-    this.setState({
+  const onValueEnter = (event: TooltipAreaEvent) => {
+    setAreaState({
       activePoint: event.pointX,
       activeValues: event.value
     });
-  }
-
-  onValueLeave() {
-    this.setState({
-      activePoint: undefined,
-      activeValues: undefined
-    });
-  }
-
-  renderArea(data: ChartInternalShallowDataShape[], index = 0) {
-    const {
-      id,
-      width,
-      xScale,
-      yScale,
-      area,
-      line,
-      interpolation,
-      animated
-    } = this.props;
+  };
 
-    return (
-      <Fragment>
-        {line && (
-          <CloneElement<LineProps>
-            element={line}
-            xScale={xScale}
-            yScale={yScale}
-            data={data}
-            width={width}
-            index={index}
-            hasArea={area !== null}
-            animated={animated}
-            interpolation={interpolation}
-            color={this.getColor.bind(this)}
-          />
-        )}
-        {area && (
-          <CloneElement<AreaProps>
-            element={area}
-            id={`${id}-area-${index}`}
-            xScale={xScale}
-            yScale={yScale}
-            data={data}
-            index={index}
-            animated={animated}
-            interpolation={interpolation}
-            color={this.getColor.bind(this)}
-          />
-        )}
-      </Fragment>
-    );
-  }
-
-  renderSymbols(data: ChartInternalShallowDataShape[], index = 0) {
-    const {
-      xScale,
-      yScale,
-      symbols,
-      id,
-      height,
-      width,
-      animated,
-      area
-    } = this.props;
-    const { activeValues } = this.state;
-
-    const visible = symbols !== null;
-    const activeSymbols =
-      (symbols && symbols.props.activeValues) || activeValues;
-
-    // Animations are only valid for Area
-    const isAnimated = area !== undefined && animated && !activeSymbols;
+  const onValueLeave = () => {
+    setAreaState(initialAreaState);
+  };
 
-    return (
-      <Fragment>
-        {visible && (
-          <CloneElement<PointSeriesProps>
-            element={symbols}
-            key={`point-series-${id}`}
-            id={id}
-            height={height}
-            width={width}
-            activeValues={activeSymbols}
-            xScale={xScale}
-            yScale={yScale}
-            index={index}
-            data={data}
-            animated={isAnimated}
-            color={() => this.getColor(data, index)}
-          />
-        )}
-      </Fragment>
-    );
-  }
+  const renderArea = useCallback(
+    (data: ChartInternalShallowDataShape[], index = 0) => {
```

Comment:
This is called data and the arg is called data. Should be renamed.

Reply:
fixed